controlplane/kubeadm/internal/cluster_test.go

```diff
@@ -178,6 +154,178 @@ func TestGetMachinesForCluster(t *testing.T) {
 	g.Expect(machines).To(HaveLen(1))
 }
 
+func TestGetWorkloadCluster(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	// create an envtest for all subtests
+	testEnv := helpers.NewTestEnvironment()
```

Comment:
Glad to see you are starting to follow test guidelines even before they are approved (I hope discussion wont impact your PR).
However, It is not sustainable to have a testenv for each test; the simplest solution now is to have one for each package: below a possible way to implement this:

```golang
var (
	testEnv *helpers.TestEnvironment
	ctx     = context.Background()
)

func TestMain(m *testing.M) {
	// TODO: consider if to use a singleton so the test environment is shared among test suites

	// Bootstrapping test environment
	testEnv = helpers.NewTestEnvironment()
	go func() {
		if err := testEnv.StartManager(); err != nil {
			panic(fmt.Sprintf("Failed to start the envtest manager: %v", err))
		}
	}()
	// Run tests
	code := m.Run()
	// Tearing down the test environment
	// TODO: it seems that etcd is not properly killed
	if err := testEnv.Stop(); err != nil {
		panic(fmt.Sprintf("Failed to stop the envtest: %v", err))
	}

	// Report exit code
	os.Exit(code)
}
```

Reply:
Yeah..the reason I did this PR was so that I could better understand the envtest impact and provide better feedback to your proposal! 😄 
Also I haven't used envtest before so I was just trying to learn its usage.

This is cool..I haven't used `TestMain` before!!